### PR TITLE
Set targeting

### LINKED ISSUE #6: Develop Set targeting
Enemies will fire in set directions, without factoring in the player. These will act as projectiles that the player cannot account for, and will test the reactions of the player, as projectiles could be in unexpected places, such as around corners or immediately in the direction the player wants to move in.

### PR DESCRIPTION
**Pull request by Sirkular**
Refactors firing mods to belong to each weapon
closes #6

### CODE REVIEW

File: Assets/Scripts/Globals.cs [hunk 1]
```diff
@@ -81,7 +81,8 @@ public enum EnemyType
 public enum EnemyVariantType
 {
     NONE,
-    HEALER
+    HEALER,
+	SET
```
Comment (Sirkular) [inline Assets/Scripts/Globals.cs:85]: This doesn't look different on my screen... then I realized, someone's using FOUR SPACES instead of a tab. MAX?!

Comment (MaXeraph) [reply Assets/Scripts/Globals.cs:85]: damn who would do such a thing

Comment (MaXeraph) [inline Assets/Scripts/Globals.cs:85]: pls dont commit ununiformed indentations pls

File: Assets/Scripts/Enemy/EnemySpawn.cs [hunk 2]
```diff
@@ -193,6 +193,12 @@ void SpawnEnemy(EnemyType enemy)
             variant = EnemyVariantType.NONE;
             enemyType = enemy;
         }
+
+        random = Random.Range(0, 10);
+        if (enemyType == EnemyType.TURRET && random < 5)
+        {
+            variant = EnemyVariantType.SET;
+        }
```
Comment (MaXeraph) [inline Assets/Scripts/Enemy/EnemySpawn.cs:201]: not a nitpick but at somepoint we might have to do something about this probability spawning

File: Assets/Scripts/Enemy/EnemyFactory.cs [hunk 3]
```diff
@@ -104,5 +104,13 @@ void AddTurretToRoster()
 
         // Define enemyInfo for each type.
         _enemyInfo.Add(EnemyType.TURRET, new EnemyInfo(EnemyMovements.TurretMovement, Globals.DirectTargeting, EnemyMovements.TurretSetup, bulletBehaviour, fireRate));
+        
+        _enemyPostSetups.Add(EnemyVariantType.SET, CreateSetTurretVariant);
+    }
+
+    void CreateSetTurretVariant(EnemyBehaviour eb)
```
Comment (MaXeraph) [inline Assets/Scripts/Enemy/EnemyFactory.cs:111]: Are we explicitly saying that Set Targetting is only for Turrets ? I would probably 
```suggestion
    void CreateSetDirectionVariant(EnemyBehaviour eb)
```

**Review by MaXeraph — APPROVED** [review]

**Review by zhuandy42 — APPROVED** [review]

**Pull request merged**